contracts/express/AxelarExpressExecutable.sol

```diff
@@ -2,33 +2,29 @@
 
 pragma solidity ^0.8.0;
 
-import { IAxelarGateway } from '../interfaces/IAxelarGateway.sol';
+import { AxelarGMPExecutableWithToken } from '../executable/AxelarGMPExecutableWithToken.sol';
+import { IAxelarGMPExecutable } from '../interfaces/IAxelarGMPExecutable.sol';
+import { IAxelarGMPExecutableWithToken } from '../interfaces/IAxelarGMPExecutableWithToken.sol';
+import { IAxelarGMPGatewayWithToken } from '../interfaces/IAxelarGMPGatewayWithToken.sol';
 import { ExpressExecutorTracker } from './ExpressExecutorTracker.sol';
-
 import { SafeTokenTransferFrom, SafeTokenTransfer } from '../libs/SafeTransfer.sol';
 import { IERC20 } from '../interfaces/IERC20.sol';
 
-contract AxelarExpressExecutable is ExpressExecutorTracker {
+abstract contract AxelarExpressExecutable is ExpressExecutorTracker, AxelarGMPExecutableWithToken {
     using SafeTokenTransfer for IERC20;
     using SafeTokenTransferFrom for IERC20;
 
-    IAxelarGateway public immutable gateway;
-
-    constructor(address gateway_) {
-        if (gateway_ == address(0)) revert InvalidAddress();
-
-        gateway = IAxelarGateway(gateway_);
-    }
+    constructor(address gateway_) AxelarGMPExecutableWithToken(gateway_) {}
 
     function execute(
         bytes32 commandId,
         string calldata sourceChain,
         string calldata sourceAddress,
         bytes calldata payload
-    ) external {
+    ) external override(AxelarGMPExecutableWithToken, IAxelarGMPExecutable) {
         bytes32 payloadHash = keccak256(payload);
 
-        if (!gateway.validateContractCall(commandId, sourceChain, sourceAddress, payloadHash))
+        if (!gatewayWithToken().validateContractCall(commandId, sourceChain, sourceAddress, payloadHash))
```

Comment:
use the gateway interface based on which method is needed
```suggestion
        if (!gateway().validateContractCall(commandId, sourceChain, sourceAddress, payloadHash))
```

Reply:
got it, updated.